Mathlib/RingTheory/Congruence.lean

```diff
@@ -76,27 +71,17 @@ section Basic
 
 variable [Add R] [Mul R] (c : RingCon R)
 
-/-- Every `RingCon` is also an `AddCon` -/
-def toAddCon : AddCon R :=
-  { c with }
-#align ring_con.to_add_con RingCon.toAddCon
-
-/-- Every `RingCon` is also a `Con` -/
-def toCon : Con R :=
-  { c with }
-#align ring_con.to_con RingCon.toCon
-
 --Porting note: upgrade to `FunLike`
 /-- A coercion from a congruence relation to its underlying binary relation. -/
 instance : FunLike (RingCon R) R fun _ => R → Prop :=
   { coe := fun c => c.r,
     coe_injective' := fun x y h => by
       rcases x with ⟨⟨x, _⟩, _⟩
       rcases y with ⟨⟨y, _⟩, _⟩
-      have : x = y := h
-      subst x; rfl }
+      congr!
+      rw [Setoid.ext_iff,(show x.Rel = y.Rel from h)]
+      simp}
 
-@[simp]
 theorem rel_eq_coe : c.r = c :=
```

Comment:
Why did this lose its `simp` attribute?

Reply:
LHS simplified (asymmetrically). Removed it and things seemed fine. Let me investigate further with the switching of order.

Reply:
I think that's probably expected (and one of the things I dislike about new-style structures), though perhaps worth a comment